hw/ip/rv_core_ibex/data/rv_core_ibex_testplan.hjson

```diff
@@ -0,0 +1,327 @@
+      milestone: V1
+      test: ["irq"]
+    }
+    {
+      name: interrupt_nested_test
+      desc: '''
+            - randomly assert interrupt lines during program execution
+            - verify that jumps to interrupt handler
+            - trigger another interrupt during first intr_handler execution
+            - verify that core takes this nested interrupt and handles it, then restores previous interrupt
+              state and finishes executing first interrupt handler - ibex implements nonstandard MSTACK csrs to
+              save/restore state in case of nested interrupt'''
+      milestone: V2
+      test: ["irq"]
+    }
+    {
+      name: interrupt_priority_test
+      desc: '''
+            - randomly assert multiple interrupt lines at once
+            - verify that core jumps to interrupt handler corresponding to the highest priority interrupt (will
+              require multiple interrupt handler, one for each type)'''
+      milestone: V2
+      test: ["irq"]
+    }
+    {
+      name: csr_test
+      desc: '''
+            Perform all CSR instructions to implemented privileged CSR
+            Verify the reset value of the privileged CSR
+            Verify WARL field can be upated properly'''
+      milestone: V1
+      test: ["csr"]
+    }
+    {
+      name: reset_test
+      desc: '''
+            Reset the core a random number of times during program execution, core
+            should jump back to start address and re-enter the program.
+            Ensure to flush all testbench state to prevent contamination after reset.'''
+      milestone: V1
+      test: ["reset"]
+    }
+    {
+      name: perf_counter_test
+      desc: '''
+            Run random instruction test, and dump performance counters for checking'''
+      milestone: V1
+      test: ["perf"]
+    }
+    {
+      name: rv32im_instr_test
+      desc: '''
+            Specify RV32IM architecture to compiler to generate target specific
+            instructions (no compressed instructions).'''
+      milestone: V1
+      test: ["rand_instr"]
+    }
+    {
+      name: user_mode_rand_test
+      desc: '''
+            Fully randomized RV32IMC instructions, boot in U-mode,
+            test should finish successfully'''
+      milestone: V2
+      test: ["priv_mode"]
+    }
+    {
+      name: umode_tw_test
+      desc: '''
+            Set mstatus.tw, and enable random generation of WFI instructions.
+            Upon encountering WFI in U-mode, core should trap to M-mode
+            illegal instruction exception handler.'''
+      milestone: V2
+      test: ["priv_mode"]
+    }
+    {
+      name: debug_ebreaku_test
+      desc: '''
+            - set dcsr.ebreaku at begining of test
+            - randomly insert ebreak instructions in generated code
+            - boot core into U-mode
+            - verify that ebreak instructions from U-mod now enter debug mode'''
+      milestone: V2
+      test: ["debug"]
+    }
+  ]
+}
```

Comment:
Add a TODO for PMP mode test

Reply:
I don't think HJSON supports comments, but I can add a dummy entry with TODO information.

Reply:
*H*JSON supports all types of comments: https://hjson.org/#comments

Reply:
Thanks for the clarication @imphil! I'll  update the PR accordingly.

Reply:
The testplan is meant to list ALL tests whether written or not. Rather than a TODO, I will suggest specifying it fully so that it shows up as an entry in the table indicating that you are indeed planning to develop it.

Reply:
@sriyerg - Sure, in that case I will prepare testplan entries for all the PMP tests I plan to implement and will update them in this file. Thanks for the feedback!

Reply:
@udinator, are you updating the TODO in this PR or a subsequent one (which is also fine)?

Reply:
I am working on testplanning for the PMP, but since this PR has been open for a while, I think it is fine if this one gets merged, and then I open up the PMP testplan PR separately.

Reply:
SGTM.